widget/window_switcher.lua

```diff
@@ -0,0 +1,350 @@
+    awful.keygrabber.stop(window_switcher_grabber)
+    window_switcher_box.visible = false
+end
+
+local function draw_widget(type, background, border_width, border_radius, border_color, clients_spacing, client_icon_horizontal_spacing, client_width, client_height, client_margins, thumbnail_margins, name_margins, name_valign, name_forced_width, name_font, name_normal_color, name_focus_color, icon_valign, icon_width, mouse_keys)
+    local tasklist_widget = type == "thumbnail"
+    and
+        awful.widget.tasklist {
+            screen = awful.screen.focused(),
+            filter = awful.widget.tasklist.filter.currenttags,
+            buttons = mouse_keys,
+            style = { font = name_font, fg_normal = name_normal_color, fg_focus = name_focus_color },
+            layout = { layout  = wibox.layout.flex.horizontal, spacing = clients_spacing },
+            widget_template =
+            {
+                widget = wibox.container.background,
+                id = "bg_role",
+                forced_width = client_width,
+                forced_height = client_height,
+                create_callback = function(self, c, _, __)
+                    local content = gears.surface(c.content)
+                    local cr = cairo.Context(content)
+                    local x, y, w, h = cr:clip_extents()
+                    local img = cairo.ImageSurface.create(cairo.Format.ARGB32, w - x, h - y)
+                    cr = cairo.Context(img)
+                    cr:set_source_surface(content, 0, 0)
+                    cr.operator = cairo.Operator.SOURCE
+                    cr:paint()
+                    self:get_children_by_id("thumbnail")[1].image = gears.surface.load(img)
+                end,
+                {
+                    {
+                        {
+                            horizontal_fit_policy = "fit",
+                            vertical_fit_policy = "fit",
+                            id = "thumbnail",
+                            widget = wibox.widget.imagebox
+                        },
+                        margins = thumbnail_margins,
+                        widget = wibox.container.margin
+                    },
+                    {
+                        {
+                            {
+                                id     = "icon_role",
+                                widget = wibox.widget.imagebox
+                            },
+                            forced_width = icon_width,
+                            valign = icon_valign,
+                            widget = wibox.container.place
+                        },
+                        {
+                            {
+                                forced_width = name_forced_width,
+                                valign  = name_valign,
+                                id = "text_role",
+                                widget = wibox.widget.textbox
+                            },
+                            margins = name_margins,
```

Comment:
I was mentioning that the text started to disappear on stable a couple weeks ago (sorry for that :sweat_smile:) and it's because of the margins I think: If I comment that line out, the text does appear on stable. Maybe just do that as a quick fix? I failed to see why that is but even if I set all margins to 0 in the `name_margins` initalization then it still doesn't appear.